From cc3f93618d828949c7c9cbad2afc33f5b0ea42ce Mon Sep 17 00:00:00 2001
From: Salvatore Bonaccorso <carnil@debian.org>
Date: Sat, 27 Mar 2021 09:19:45 +0100
Subject: [PATCH] libsas: Avoid ABI change for removal of notifier indirection

121181f3f839 ("scsi: libsas: Remove notifier indirection") cleaned up
the code to reduce function pointer usages calling notify_port_event and
notify_phy_event.

Keep those in struct sas_ha_struct even though they won't be used
anymore.

Signed-off-by: Salvatore Bonaccorso <carnil@debian.org>
---
 include/scsi/libsas.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/scsi/libsas.h b/include/scsi/libsas.h
index e6a43163ab5b..473b27f5c219 100644
--- a/include/scsi/libsas.h
+++ b/include/scsi/libsas.h
@@ -391,6 +391,10 @@ struct sas_ha_struct {
 	int strict_wide_ports; /* both sas_addr and attached_sas_addr must match
 				* their siblings when forming wide ports */
 
+	/* LLDD calls these to notify the class of an event. */
+	int (*notify_port_event)(struct asd_sas_phy *, enum port_event);
+	int (*notify_phy_event)(struct asd_sas_phy *, enum phy_event);
+
 	void *lldd_ha;		  /* not touched by sas class code */
 
 	struct list_head eh_done_q;  /* complete via scsi_eh_flush_done_q */
-- 
2.31.0

